### ISSUE TITLE
Device Skipped/Unknown Capabilities - Output

### ISSUE THREAD

**Issue by fireheadman** [pdlove/homebridge-smartthings#139]
Believe you were requesting this type of output...

This device is the **IRIS Smart Button** ([https://www.amazon.com/Iris-Smart-Home-Management-3460-L1/dp/B01GCKAP3I](https://www.amazon.com/Iris-Smart-Home-Management-3460-L1/dp/B01GCKAP3I))
2/1/2017, 12:07:38 AM] [SmartThings] Device Skipped - Group unknown, Name Baby Room Light, ID ff14e4e6-38c6-42bc-a281-ab9826f9d92e, JSON: {"name":"Baby Room Light","basename":"Iris Smart Button","deviceid":"ff14e4e6-38c6-42bc-a281-ab9826f9d92e","status":"ACTIVE","manufacturerName":"CentraLite","modelName":"3460-L","lastTime":"2017-02-01T05:16:04+0000","capabilities":{"Battery":1,"Configuration":1,"Refresh":1,"Button":1,"Sensor":1,"Actuator":1,"Holdable Button":1},"commands":{"configure":[],"refresh":[],"enrollResponse":[]},"attributes":{"battery":88,"button":"pushed","numberOfButtons":1}}

These capabilities are related to on of the following devices: 
**IRIS Smart Button, Sonos Speaker, FirstAlert ZCOMBO, Monoprice 4-N-1 Sensor**
[2/1/2017, 12:07:38 AM] [SmartThings] Unknown Capabilities: ["Relay Switch","Tamper Alert","Health Check","Button","Holdable Button","Music Player","Illuminance Measurement","Smoke Detector","Carbon Monoxide Detector"]

**Comment by fireheadman** [comment]
UPDATE:    Was going to edit my post and remove half of it, but thought it might have value to serve as a reminder for anyone else seeing these issues...

I updated the JSON API code in ST IDE (SmartApps)
BUT YOU also MUST update homebridge-smartthings.   Once I did this, it resolved the "Device Skipped" messages.

I do however still have a Unknown Capabilities showing up
[2/1/2017, 12:19:47 AM] [SmartThings] Unknown Capabilities: ["Relay Switch","Tamper Alert","Health Check","Button","Holdable Button","Music Player","Illuminance Measurement","Smoke Detector","Carbon Monoxide Detector"]

**Comment by fireheadman** [comment]
Darn it... So the update made the IRIS Smart Button (Device Skipped) issue go away, but now in the HOME App it shows up like this:
![image](https://cloud.githubusercontent.com/assets/19847195/22498412/7cb2e398-e815-11e6-84c7-03533785c243.png)
![image](https://cloud.githubusercontent.com/assets/19847195/22498416/8648f55a-e815-11e6-8f7f-60d354eaadc8.png)

**Comment by pdlove** [comment]
It's having issues because the device is coming into HomeKit as a Battery which is something Home just doesn't understand.
I'm trying to get clarification on how the button flag works on the button capability type. I've never seen one set to anything but Pushed.

**Comment by mattstocum** [comment]
I'm having the same issue with a HomeSeer WS100+ Switch. It's a switch that is also a button controller.

    [SmartThings] Device Skipped - Group  button, Name Bedroom Light, ID 130446e2-25aa-413f-b393-f4496e98c41c, JSON: {"name":"Bedroom Light","basename":"WS100+ Switch","deviceid":"130446e2-25aa-413f-b393-f4496e98c41c","status":"ACTIVE","manufacturerName":"HomeSeer Technologies","modelName":null,"lastTime":"2017-02-04T13:55:08+0000","capabilities":{"Switch":1,"Polling":1,"Configuration":1,"Refresh":1,"Indicator":1,"Button":1,"Sensor":1,"Actuator":1},"commands":{"on":[],"off":[],"poll":[],"configure":[],"refresh":[],"indicatorWhenOn":[],"indicatorWhenOff":[],"indicatorNever":[],"tapUp2":[],"tapDown2":[],"tapUp3":[],"tapDown3":[],"holdUp":[],"holdDown":[]},"attributes":{"switch":"off","indicatorStatus":"when off","button":"pushed","numberOfButtons":6}}

**Comment by sidjohn1** [comment]
I did the same as mattstocum and commented out the code that assigns the button deviceGroup in accessories/smartthings.js

In my case it's a custom mimolite device handler

**Comment by rex555** [comment]
Anything here? I am getting this after adding Cree bulbs None of them show up. My Switch still shows up, just nothing else

**Comment by pdlove** [comment]
@rex555 Sorry, I didn't see your comment on this one. What device handler are your cree bulbs using? I have several Cree Bulbs that are working just fine but their main capability is "Switch Level". They also present the "Health Check" and "Light" capabilities, but this plugin is using switch level to control/show the items.

**Comment by rex555** [comment]
Hmm interesting, using switch capabilities does not allow for dimming features though, right? Let me try using as switches instead of lights. I will report back on how that works.

**Comment by pdlove** [comment]
"Switch Level" does. It might be that new additions only get Lights. I'll review any of the new capabilities or ones I've previously ignored over the next week.

**Comment by rex555** [comment]
I finally got all this working, and it turns out it was my own stupidity (well, really Smartthings shitty app, but hey) that caused my issues.

Turns out, when you add things, you DON'T go to "Marketplace" and select "SmartApps". At the bottom of the list, select "My Apps" and Select "JSON Complete API" from the list again. That is what I was doing.

Turns out that gives you another instance of the "JSON Complete API" in the place you are SUPPOSED to go which is the "Automation" tab on the bottom nav bar, then the "SmartApps" tab in that. I have 10-15 "JSON Complete API" entries there, each with their OWN app_id and access_token! so after cleaning all but one out, and re-copying the IDs to the config.json on the homebridge server, lo and behold I have control!!

My only suggestion for you is maybe a section on how to add more things that points people to the right place to do that. Or maybe I am just a big dummy LOL

Thanks for your work, it is immensely helpful and saves me writing it all myself.
